### PR TITLE
Feature 1566 - Add merge/update branch support

### PR DESCRIPTION
**Pull request by abelpz**
For testing this you will need to make changes to both master and user branch,
This was my method:

- Open tCC and select an org, repo and .tsv file.
- Check all the merge/update buttons to see what they do on first print of the app.
- Start making a change to the file
- Check all the merge/update buttons to see what they do
- Click save
- Check all the merge/update buttons to see what they do
- and so on..

For testing what happens when there are changes in master:

Click the compare target button in tCC which will take you to the right repo on DCS:
tcc-merge-update-1

Then make sure you select the master branch, and start making changes to the same file you starting working at in tCC and others, you may need to refresh tCC or make new changes in tCC to trigger a new state update for the merge/update buttons.

### CODE REVIEW

File: src/components/branch-merger/components/MergeBranchButton.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import IconButton from '@material-ui/core/IconButton';
+import Badge from '@material-ui/core/Badge';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { Tooltip } from '@material-ui/core';
+import { MdOutlinePublish, MdPublish } from "react-icons/md";
+
+export function MergeBranchButton({
+  onClick = () => undefined,
+  onMerged = () => undefined,
```
Comment (jincypjose) [inline src/components/branch-merger/components/MergeBranchButton.js:10]: Do we need this prop?

Comment (abelpz) [reply src/components/branch-merger/components/MergeBranchButton.js]: Good catch, I have removed it. Thanks!

File: src/components/branch-merger/components/UpdateBranchButton.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { MdUpdate, MdUpdateDisabled } from 'react-icons/md';
+import IconButton from '@material-ui/core/IconButton';
+import Badge from '@material-ui/core/Badge';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { Tooltip } from '@material-ui/core';
+
+export function UpdateBranchButton({
+  onClick = () => undefined,
+  onUpdated = () => undefined,
```
Comment (jincypjose) [inline src/components/branch-merger/components/UpdateBranchButton.js:10]: Can we remove this prop?

**Review by jincypjose — APPROVED** [review]

File: src/components/branch-merger/components/UpdateBranchButton.js (hunk 2 above)
Comment (abelpz) [reply src/components/branch-merger/components/UpdateBranchButton.js]: Oh, I should have checked this one too, thanks @jincypjose, I have removed it.

**Pull request merged**